testsuite/integration/clustering/src/test/java/org/jboss/as/test/clustering/cluster/ejb/remote/TransactionalRemoteStatelessTestCase.java

```diff
@@ -0,0 +1,450 @@
+                Assert.fail("Expected a NoSuchEJBException as transaction affinity needs to be maintained");
+            } catch (NoSuchEJBException expected) {
+                // expected as the deployment was removed from the node
+            }
+        } finally {
+            if(txn != null) txn.rollback();
+            initCtx.close();
+        }
+    }
+
+    private Incrementor getStatelessRemoteBean(InitialContext ctx, Class<? extends Incrementor> beanClass) {
+        return getRemoteBean(ctx, beanClass, false);
+    }
+
+    private Incrementor getRemoteBean(InitialContext ctx, Class<? extends Incrementor> beanClass, boolean isStateful) {
+        String lookupBeanUrl = "ejb:/"  + MODULE_NAME +"/" + beanClass.getSimpleName() + "!" + Incrementor.class.getName();
+        if(isStateful) lookupBeanUrl += "?stateful";
+
+        try {
+            return (Incrementor) ctx.lookup(lookupBeanUrl);
+        } catch (NamingException ne) {
+            throw new IllegalStateException(String.format("Cannot find ejb bean '%s' while "
+                    + "looking up for name '%s'", beanClass, lookupBeanUrl), ne);
+        }
+    }
+
+    private UserTransaction getUserTransaction(InitialContext ctx) {
+        String lookupUserTransaction = "txn:RemoteUserTransaction";
+        try {
+            return (UserTransaction) ctx.lookup(lookupUserTransaction);
+        } catch (NamingException ne) {
+            throw new IllegalStateException(String.format("Cannot look up '%s' at context: %s",
+                    ctx, lookupUserTransaction), ne);
+        }
+    }
+
+    private InitialContext getInitialContext(InitialContextLookupType lookupType, String nodeHostName) {
+        final Hashtable<String, String> jndiProperties = new Hashtable<>();
+        jndiProperties.put(Context.URL_PKG_PREFIXES, "org.jboss.ejb.client.naming");
+        jndiProperties.put(Context.INITIAL_CONTEXT_FACTORY, "org.wildfly.naming.client.WildFlyInitialContextFactory");
+        if(nodeHostName.contains(":") && !nodeHostName.contains("[")) {
+            nodeHostName = "[" + nodeHostName + "]"; // escaping host and port for IPv6
+        }
+
+        String lookupProviderUrl = "<see wildfly-config.xml>";
+        switch(lookupType) {
+            case HTTP:
+                lookupProviderUrl = String.format("http://%s:%s/wildfly-services", nodeHostName, node0Port);
+                jndiProperties.put(Context.PROVIDER_URL, lookupProviderUrl);
+                // see server/configuration/application-users.properties
+                jndiProperties.put(Context.SECURITY_PRINCIPAL, "user1");
+                jndiProperties.put(Context.SECURITY_CREDENTIALS, "password1");
+                break;
+            case REMOTE_HTTP:
+                lookupProviderUrl = String.format("remote+http://%s:%s", nodeHostName, node0Port);
+                jndiProperties.put(Context.PROVIDER_URL, lookupProviderUrl);
+                break;
+            case WILDFLY_CONFIG:
```

Comment:
Just a nitpick, but not sure if this is necessary if the handling already is in `default`.

Reply:
I see, but I would leave it as a marker of the functionality. I think it does no harm.